### PR TITLE
MCLOUD-6898: Add option to customize port for MailHog

### PR DESCRIPTION
**Pull request by BaDos**
### Description

Added options to configure HTTP and SMTP ports for MailHog

### Fixed Issues (if relevant)

- https://jira.corp.magento.com/browse/MCLOUD-6898
- https://jira.corp.magento.com/browse/MCLOUD-6917

### Manual testing scenarios

Covered with Integration and Functional tests

### Release notes

Added MailHog configuration options to the ece-docker build:compose command to customize the HTTP and SMTP ports, and to disable MailHog. See [Set up email](https://devdocs.magento.com/cloud/docker/docker-config.html#set-up-email).

### Associated documentation updates

https://github.com/magento/devdocs/pull/7860

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: src/Test/Integration/_files/custom_cloud_base/docker-compose.exp.yml [hunk 1]
```diff
@@ -138,8 +138,8 @@ services:
     image: 'mailhog/mailhog:latest'
     restart: always
```
Comment (shiftedreality) [inline src/Test/Integration/_files/custom_cloud_base/docker-compose.exp.yml]: Please remove `restart: always` since it causes issues

Comment (BaDos) [reply src/Test/Integration/_files/custom_cloud_base/docker-compose.exp.yml]: Done.

File: src/Compose/ProductionBuilder.php [hunk 2]
```diff
@@ -402,7 +402,13 @@ public function build(Config $config): Manager
                 self::SERVICE_MAILHOG,
                 $this->serviceFactory->create(
                     ServiceInterface::SERVICE_MAILHOG,
-                    $this->serviceFactory->getDefaultVersion(ServiceInterface::SERVICE_MAILHOG)
+                    $this->serviceFactory->getDefaultVersion(ServiceInterface::SERVICE_MAILHOG),
+                    [
+                        'ports' => [
+                            ($config->getMailHogSmtpPort() ?: '1025').':1025',
```
Comment (shiftedreality) [inline src/Compose/ProductionBuilder.php]: Move defaults port to source where http ports located

**Review by shiftedreality — CHANGES_REQUESTED** [review]

**Review by arhiopterecs — DISMISSED** [review]

File: src/Compose/ProductionBuilder.php (hunk 2 above)
Comment (BaDos) [reply src/Compose/ProductionBuilder.php]: Done

**Review by shiftedreality — APPROVED** [review]

**Review by arhiopterecs — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**